{{ cookiecutter.__project_name_kebab }}/package.json

```diff
@@ -6,11 +6,7 @@
   "scripts": {
     "start": "webpack --config ./webpack.config.js --mode development --progress --watch",
     "build": "webpack --config ./webpack.config.js --mode production",
-    "format": "prettier --write '**/?(.)*.{css,scss,js,ts,tsx,json,yaml,yml}'",
-    "lint:js": "eslint --ext .js,.ts,.tsx --report-unused-disable-directives --max-warnings 16 ./",
-    "lint:css": "stylelint **/*.scss",
-    "lint:format": "prettier --check '**/?(.)*.{css,scss,js,ts,tsx,json,yaml,yml}'",
-    "lint": "npm run lint:js && npm run lint:css && npm run lint:format"
+    "format": "git ls-files --others --cached --exclude-standard | xargs pre-commit run --files"
```

Comment:
Still need to do a bit more testing, but this _might_ be the pre-commit coping strategy I've been looking for.

```
git ls-files --others --cached --exclude-standard
```

lists all files git knows about _including untracked files_ but excluding anything that has been `.gitignored`, the `.git` directory itself, etc. Then I'm using `xargs` to pass that whole list to `pre-commit`'s `--files` argument.

Reply:
Personally I don't see anything wrong with using npm scripts as a general purpose task runner if you have it in your project anyway. I've done it before and I'll probably do it again.

That said, I do usually use makefiles for my own projects (example of a fairly typical one for one of my own libraries https://github.com/chris48s/stage-left/blob/master/Makefile ) Also other task runners are available.


I feel like if we go down this route, exactly what should/shouldn't go in the `Makefile`, how we document it, etc is probably an issue/PR unto itself

- Should `make test` run `coverage run testmanage.py test --deprecation all -vv` or `tox`?
- Should we wrap the remaining NPM scripts (`build`, `start`) in make commands?
- Should there be a `make install`. If so, exactly what should it run?
- Do we defer to using `make` commands in tox/CI?
- etc

I think those are all questions worth thinking about (and indeed, providing an abstraction over), but I think getting bogged down in that in this PR means we'll go off on a tangent and end up blocking a chunk of work which is otherwise close to "finished" (hopefully).

So here's my suggestion: I've moved that snippet (which I think is useful) to a line in the docs. Lets make an issue for "adopt a task runner" or whatever and make that an issue for another time.

Reply:
moved to https://github.com/wagtail/cookiecutter-wagtail-package/issues/48

Reply:
I think another good reason not to rely on npm scripts as a general-purpose task runner in this specific case is that if we do https://github.com/wagtail/cookiecutter-wagtail-package/issues/14 (which I think we should do at some point, not all wagtail packages require frontend/js), npm won't necessarily be there for all generated packages.